src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4876,6 +4876,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           case LookupInaccessible(sym, msg) => issue(AccessError(tree, sym, context, msg))
           case LookupNotFound               =>
             inEmptyPackage orElse lookupInRoot(name) match {
+              // ignore wildcard bodies inserted by the Java parser
+              case NoSymbol if context.unit.isJava && name == termNames.WILDCARD => tree
```

Comment:
Can we just skip type checking the body all together for java definitions since we know it'll always be `_`?